pyvo/dal/sia.py

```diff
@@ -503,12 +503,21 @@ def format(self):
 
     @format.setter
     def format(self, format_):
+        print("Set format")
         setattr(self, "_format", format_)
 
-        if isinstance(format_, (str, bytes)):
-            format_ = [format_]
-
-        self["FORMAT"] = ",".join(_.upper() for _ in format_)
+        if not isinstance(format_, list):
+            format_ = format_.split(',')
+        normalized_formats = []
+        for user_input in format_:
+            if user_input.upper() in ['ALL', 'METADATA', 'GRAPHIC', 'GRAPHIC-ALL']:
+                normalized_formats += [user_input.upper()]
```

Comment:
This is a bit bike-shedding as normalized_formats will never be a very long list where we need to optimize on whether a new copy is made or not or how many intermediate list instances are created, nevertheless I would  think the following is a cleaner logic, here and below, too:
```suggestion
                normalized_formats.append(user_input.upper)
```